src/repository/fetcher.ts

```diff
@@ -0,0 +1,29 @@
+import { EventEmitter } from 'events';
+import { ClientFeaturesDelta, ClientFeaturesResponse } from '../feature';
+import { Mode } from '../unleash-config';
+
+export interface FetcherInterface extends EventEmitter {
+  start(): Promise<void>;
+  stop(): void;
+}
+
+export interface FetchingOptions {
```

Comment:
replace with 2 narrowly focused interfaces for polling and streaming to avoid coupling